examples/espressif/port/src/iotc_bsp_io_net_posix.c

```diff
@@ -0,0 +1,289 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iotc_bsp_io_net.h>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <stdio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef MAX
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+#endif
+
+iotc_bsp_io_net_state_t iotc_bsp_io_net_create_socket(
+    iotc_bsp_socket_t* iotc_socket) {
+  *iotc_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+  if (-1 == *iotc_socket) {
+    return IOTC_BSP_IO_NET_STATE_ERROR;
+  }
+
+  /* Enable nonblocking mode for a posix socket */
+  const int flags = fcntl(*iotc_socket, F_GETFL, 0);
+
+  if (-1 == flags || -1 == fcntl(*iotc_socket, F_SETFL, flags | O_NONBLOCK)) {
+    return IOTC_BSP_IO_NET_STATE_ERROR;
+  }
+
+  return IOTC_BSP_IO_NET_STATE_OK;
+}
+
+iotc_bsp_io_net_state_t iotc_bsp_io_net_connect(iotc_bsp_socket_t* iotc_socket,
```

Comment:
This should be adapted to [the new version](https://github.com/GoogleCloudPlatform/iot-device-sdk-embedded-c/blob/master/include/bsp/iotc_bsp_io_net.h#L239).

Reply:
As it seems the port is based on the POSIX port (file name keeps the posix string). You might give a try to update these files with the current posix port but probably you have to keep an eye the specialities this "posix" version has.